### DISCUSSION TITLE
Convert dtype='object' time to datetime64

### DISCUSSION THREAD

**Discussion by zxdawn** [pydata/xarray#6708] [Q&A] [answered]
I have the same problem as this [question](https://stackoverflow.com/q/62572678/7347925) on stack overflow. To convert the time whose type is "object" to datetime64, we need to run `ds['time'] = pd.DatetimeIndex(ds['time'].values)` twice. Any idea why it should be run twice and is there a better method?

**Comment by keewis** [comment, marked as answer]
Usually I'd try using
```python
ds["time"] = ds["time"].astype("datetime64[ns]")
```

If that does not work, could you post a example I can use to reproduce your issue?

**Reply by zxdawn** [reply]
Thanks, it works well. In case anyone wants to check it, here's the example:
```
import numpy as np
import xarray as xr

data = np.ones((360,720))
lats = np.arange(-89.75, 90, 0.5) * -1
lons = np.arange(-179.75, 180, 0.5)
coords =  {'lat': lats, 'lon':lons}

da = xr.DataArray(data, coords=coords)
da = da.expand_dims('time').assign_coords(time=('time', [pd.Timestamp('2019-08-11 01:55:21.510076928+00:00')]))

# old method
# da['time'] = pd.DatetimeIndex(da['time'].values)
# da['time'] = pd.DatetimeIndex(da['time'].values)

# new method
da["time"] = da["time"].astype("datetime64[ns]")
```
